src/otx/engine/engine.py

```diff
@@ -514,12 +514,15 @@ def explain(
             ckpt_path=ckpt_path,
         )
 
+        explain_hook = self.trainer.model.model.explain_hook
         # Optimize for memory <- TODO(negvet)
-        raw_saliency_maps = self.trainer.model.model.explain_hook.records
+        raw_saliency_maps = explain_hook.records
+
         return get_processed_saliency_maps(
             raw_saliency_maps,
```

Comment:
In this case we can get `raw_saliency_maps` inside of `get_processed_saliency_maps`, don't we?

Reply:
Good point, moved. Didn't add if statement; 
```
if `MaskRCNN`:
  explain_hook.func(predictions)
else:
  explain_hook.records
```
to make code more simple, since `explain_hook.records` is preferable path